librz/bin/format/le/le.c

```diff
@@ -57,828 +87,1749 @@ const char *le_get_arch(rz_bin_le_obj_t *bin) {
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->fixupsize) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->fixupsum) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->ldrsize) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->ldrsum) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->objtab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->objcnt) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->objmap) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->itermap) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->rsrctab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->rsrccnt) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->restab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->enttab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->dirtab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->dircnt) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->fpagetab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->frectab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->impmod) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->impmodcnt) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->impproc) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->pagesum) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->datapage) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->preload) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->nrestab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->cbnrestab) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->nressum) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->autodata) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->debuginfo) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->debuglen) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->instpreload) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->instdemand) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->heapsize) ||
+		!rz_buf_read_le32_offset(bin->buf, &off, &bin->header->stacksize)) {
+		free(bin->header);
+		return false;
+	}
+
+	if (bin->header->border || bin->header->worder) {
+		// shouldn't be hard to support, but I couldn't find any such binaries to test on
+		RZ_LOG_ERROR("LE: only little-endian byte and word order is supported, "
+			     "got (%d, %d), expected (0, 0).\n",
+			bin->header->border, bin->header->worder);
+		return false;
+	}
+	bin->is_le = !memcmp("LE", bin->header->magic, 2);
+
+	return true;
+}
+
+// Loading objects, see [1] 3.4 Object Table
+static bool le_load_objects(rz_bin_le_obj_t *bin) {
+	LE_header *h = bin->header;
+	LE_object *objects = NULL;
+	if (!h->objcnt) {
+		return true; // no objects, binary is a forwarders-only library
+	}
+	if (false) {
```

Comment:
Unnecessary block

Reply:
Which one? objcnt is needed to make debug build calloc happy when there's no objects. false is used by multiple CHECKs below.